### PR TITLE
Feature/Birch

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#17 
#### What does this implement/fix? Explain your changes.
* Birch Model support
* CFNode Transporter
* CFSubCluster Transporter

### CODE REVIEW

File: pymilo/transporters/cfnode_transporter.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                        "pymilo_model_type": "_CFNode",
+                        "pymilo_cfnode_value": "PYMILO_CFNODE_RECURSION",
+                        "pymilo_cfnode_id": value_id,
+                    }
+                else:
+                    data[key] = {
+                        "pymilo_model_type": "_CFNode",
+                        "pymilo_cfnode_value": self.serialize_cfnode(value, gdst),
+                        "pymilo_cfnode_id": value_id,
+                    }
```
Comment (sadrasabouri) [inline pymilo/transporters/cfnode_transporter.py:96]: How about this:?
```python
data[key] = {
    "pymilo_model_type": "_CFNode",
    "pymilo_cfnode_value": "PYMILO_CFNODE_RECURSION" if value_id in self.all_cfnodes else self.serialize_cfnode(value, gdst),
    "pymilo_cfnode_id": value_id,
}
```

File: pymilo/transporters/cfnode_transporter.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                            self.serialize_cfsubcluster(cf_subcluster, gdst) for cf_subcluster in value], }
+                else:
+                    data[key] = gdst.serialize(data, key, str(_CFNode))  # TODO model name
+            else:
+                data[key] = gdst.serialize(data, key, str(_CFNode))
+        return data
+
+    def deserialize_cfnode(self, cfnode_pymiloed_obj, gdst):
+        """
+        Derialize given serialized object of _CFnode class recursively.
```
Comment (sadrasabouri) [inline pymilo/transporters/cfnode_transporter.py:110]: Typo here in `Deserialize`.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

File: pymilo/utils/util.py [hunk 3]
```diff
@@ -106,3 +107,19 @@ def import_function(module_name, function_name):
     module = importlib.import_module(module_name)
     function = getattr(module, function_name)
     return function
+
+
+def has_named_parameter(func, param_name):
+    """
+    Check whether the given functino has a parameter named param_name or not.
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply pymilo/utils/util.py]: applied

File: pymilo/transporters/cfnode_transporter.py (hunk 1 above)
Comment (AHReccese) [reply pymilo/transporters/cfnode_transporter.py]: thanks, applied.

File: pymilo/transporters/cfnode_transporter.py (hunk 2 above)
Comment (AHReccese) [reply pymilo/transporters/cfnode_transporter.py]: applied.

**Review by sadrasabouri — APPROVED** [review]

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**